compute/main.go

```diff
@@ -0,0 +1,596 @@
+package main
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/compute/armcompute"
+	"github.com/Azure/azure-sdk-for-go/sdk/network/armnetwork"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/armcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resources/armresources"
+	"github.com/Azure/azure-sdk-for-go/sdk/to"
+)
+
+var subscriptionId string
+
+const (
+	resourceGroupName = "sample-resource-group"
+	vmName            = "sample-vm"
+	vnetName          = "sample-vnet"
+	subnetName        = "sample-subnet"
+	nsgName           = "sample-nsg"
+	nicName           = "sample-nic"
+	diskName          = "sample-disk"
+	publicIPName      = "sample-public-ip"
+	location          = "westus2"
+)
+
+func main() {
+	subscriptionId = os.Getenv("AZURE_SUBSCRIPTION_ID")
+	if len(subscriptionId) == 0 {
+		log.Fatal("AZURE_SUBSCRIPTION_ID is not set.")
+	}
+	//create virtual machine
+	createVM()
+
+	keepResource := os.Getenv("KEEP_RESOURCE")
+	if len(keepResource) == 0 {
+		//delete virtual machine
+		deleteVM()
+	}
+}
+
+func createVM() {
+	conn, err := connectionAzure()
+	if err != nil {
+		log.Fatalf("cannot connection Azure:%+v", err)
+	}
+	ctx := context.Background()
+
+	log.Println("start creating virtual machine...")
+	resourceGroup, err := createResourceGroup(ctx, conn)
+	if err != nil {
+		log.Fatalf("cannot create resource group:%+v", err)
+	}
+	log.Printf("Created resource group: %s", *resourceGroup.ID)
+
+	virtualNetwork, err := createVirtualNetwork(ctx, conn)
+	if err != nil {
+		log.Fatalf("cannot create virtual network:%+v", err)
+	}
+	log.Printf("Created virtual network: %s", *virtualNetwork.ID)
+
+	subnet, err := createSubnets(ctx, conn, vnetName, subnetName)
+	if err != nil {
+		log.Fatalf("cannot create subnet:%+v", err)
+	}
+	log.Printf("Created subnet: %s", *subnet.ID)
+
+	publicIP, err := createPublicIP(ctx, conn, publicIPName)
+	if err != nil {
+		log.Fatalf("cannot create public IP address:%+v", err)
+	}
+	log.Printf("Created public IP address: %s", *publicIP.ID)
+
+	// network security group
+	nsg, err := createNetworkSecurityGroup(ctx, conn, nsgName)
+	if err != nil {
+		log.Fatalf("cannot create network security group:%+v", err)
+	}
+	log.Printf("Created network security group: %s", *nsg.ID)
+
+	subnetID := subnet.ID
+	publicID := publicIP.ID
+	networkSecurityGroupID := nsg.ID
+	netWorkInterface, err := createNetWorkInterface(ctx, conn, subnetID, publicID, networkSecurityGroupID)
```

Comment:
Can you directly input subnet.ID into createNetWorkInterface()?   Why do we need to call   subnetID := subnet.ID